File: Leetcode/Python/1550. Three Consecutive Odds/solution.py
# https://leetcode.com/problems/three-consecutive-odds/description

class Solution:
    def threeConsecutiveOdds(self, arr: List[int]) -> bool:
        for idx in range(len(arr) - 2):
            for i in range(idx, idx + 3):
                if arr[i] % 2 == 0:
                    break
            else:
                return True
        
        return False

File: Leetcode/Typescript/1550. Three Consecutive Odds/solution.ts
// https://leetcode.com/problems/three-consecutive-odds/description

function threeConsecutiveOdds(arr: number[]): boolean {
  for (let idx = 0; idx < arr.length - 2; idx++) {
    let isOdd = true;

    for (let i = idx; i < idx + 3; i++) {
      if (arr[i] % 2 === 0) {
        isOdd = false;
        break;
      }
    }

    if (isOdd) {
      return true;
    }
  }

  return false;
}
